docs/dev/components/components.md

```diff
@@ -0,0 +1,94 @@
+# Components/Maven Modules
+
+Gaffer has lots of components, which can be split into different categories. This section of the developer doc provides more detail on the most important components.
+
+## Key components
+
+For more developer information on these key components, see their associated page.
+
+- [Operation](operation.md): Classes for the Operation interfaces and core operation implementations.
+- [Cache](cache.md): Classes for the Gaffer cache service.
+- [Graph](graph.md): Contains the Gaffer `Graph` object and related utilities.
+- [Data](data.md): Classes defining Gaffer's data objects: `Element`, `Edge` and `Entity`.
+- [Store](store.md): Contains the Gaffer `Store` object and related classes.
+- [Serialisation](serialisation.md): Contains classes for Serialisation in Gaffer. 
+- [Accumulo Store](accumulo-store.md): A store implemented using Apache Accumulo.
+- [Core REST](core-rest.md): Classes which provide a Gaffer REST API using Jersey/JAX-RS.
+- [Spring REST](spring-rest.md): Implementation of the Gaffer REST API deployed in a Spring Boot container.
+
+## Project Architecture (Maven)
+
+Gaffer uses Maven and from this perspective the project is made up of multiple Maven modules. All components are Maven modules, but not all modules are components as some are just parent POMs (see below).
+Maven modules are not the same as Java modules (Java 9+), which Gaffer doesn't use or support. [See here for more info on multi-module builds and the Maven reactor](https://maven.apache.org/guides/mini/guide-multiple-modules.html).
+
+Gaffer's project structure involves three kinds of module/POM, mostly based on the [Maven packaging type](https://maven.apache.org/pom.html#Packaging). We can call these:
+
+- Parent modules (`pom`)
+- JAR modules (`jar`)
+- WAR only/Demo modules (`war`).
+
+Parent modules consist of a single POM file (parent POM) which defines various properties, dependencies and settings to be inherited by JAR modules or other parent modules.
+
+JAR modules are not inherited by other modules, and in addition to the POM they contain code which is compiled into artifacts (always 'jar' and sometimes also `war`). Some contain only code used for demos.
+
+WAR only/Demo modules are not inherited by other modules, they contain only a POM and potentially config files and are used for either creating WAR archives for the REST API or are used for running demos using Maven plugins and other modules.
+
+Gaffer has 47 modules in total. When building the complete project, Maven automatically runs the build in a specific order because of dependencies between modules.
+The paths for all POMs in the Gaffer project can be seen by running `find . -name pom.xml` from the project/repository root.
```

Comment:
Does this produce significantly different output? The `tree` command is less universal than `find`. Linux distros will probably have it by default but BSD and MacOS don't have it.
A tip section could work, though as this is pretty brief it would take up more space. Usually these sections are used for saving space.

Reply:
That's fair, also noticed there is a maven plugin that can print it out which might could be mentioned `mvn org.qunix:structure-maven-plugin:modules`